src/nspawn/nspawn.c

```diff
@@ -5129,12 +5140,11 @@ static int run(int argc, char *argv[]) {
                                 goto finish;
                         }
 
-                        if (!arg_root_hash) {
-                                r = root_hash_load(arg_image, &arg_root_hash, &arg_root_hash_size);
-                                if (r < 0) {
-                                        log_error_errno(r, "Failed to load root hash file for %s: %m", arg_image);
-                                        goto finish;
-                                }
+                        r = verity_metadata_load(arg_image, arg_root_hash ? NULL : &arg_root_hash, &arg_root_hash_size,
+                                        arg_verity_data ? NULL : &arg_verity_data);
+                        if (r < 0) {
+                                r = log_error_errno(r, "Failed to read verity artefacts for %s: %m", arg_image);
```

Comment:
doesn't matter, but there's not much point in assigning the return value of the log call to `r`, given that the very same value is already in there...

Reply:
Removed - was copied from other usage in the same function